src/sentry/api/endpoints/organization_event_details.py

```diff
@@ -40,51 +41,61 @@ def get(self, request, organization, project_slug, event_id):
         event_slug = u"{}:{}".format(project.slug, event_id)
         snuba_args["conditions"].extend(get_reference_event_conditions(snuba_args, event_slug))
 
+        dataset = snuba.detect_dataset(snuba_args, aliased_conditions=True)
+
         data = serialize(event)
-        data["nextEventID"] = self.next_event_id(snuba_args, event)
-        data["previousEventID"] = self.prev_event_id(snuba_args, event)
-        data["oldestEventID"] = self.oldest_event_id(snuba_args, event)
-        data["latestEventID"] = self.latest_event_id(snuba_args, event)
+        data["nextEventID"] = self.next_event_id(snuba_args, event, dataset=dataset)
+        data["previousEventID"] = self.prev_event_id(snuba_args, event, dataset=dataset)
+        data["oldestEventID"] = self.oldest_event_id(snuba_args, event, dataset=dataset)
+        data["latestEventID"] = self.latest_event_id(snuba_args, event, dataset=dataset)
         data["projectSlug"] = project_slug
 
         return Response(data)
 
-    def next_event_id(self, snuba_args, event):
+    def next_event_id(self, snuba_args, event, dataset):
         """
         Returns the next event ID if there is a subsequent event matching the
         conditions provided. Ignores the project_id.
         """
-        next_event = eventstore.get_next_event_id(event, filter=self._get_filter(snuba_args))
+        next_event = eventstore.get_next_event_id(
+            event, filter=self._get_filter(snuba_args), dataset=dataset
```

Comment:
Should the `dataset` be part of the `filter` object? That could help consolidate some parameters and co-locate the dataset choice with the conditions.